### PR TITLE
SmallRye GraphQL non-blocking support

### PR DESCRIPTION
**Pull request by jsmrcka**
### Summary

Adds test coverage for SmallRye GraphQL:
- reactive queries
- GET queries (GraphQL over HTTP)
- `@AdaptToScalar`
- Maps support
- `@ErrorCode`

Please select the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Dependency update
- [ ] Refactoring
- [ ] Backport
- [x] New scenario (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
- [x] This change requires execution against OCP (use `run tests` phrase in comment)

### Checklist:
- [x] Methods and classes used in PR scenarios are meaningful
- [x] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

**Comment by rsvoboda** [comment]
assigned the review to @fedinskiy who worked on the initial graphql coverage

**Comment by jsmrcka** [comment]
run tests

**Review by fedinskiy — CHANGES_REQUESTED** [review]

File: http/graphql/src/test/java/io/quarkus/ts/http/graphql/GraphQLGetDisabledIT.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.QuarkusApplication;
+import io.restassured.response.Response;
+
+@QuarkusScenario
+public class GraphQLGetDisabledIT {
```
[the remark this answers is not in the document]
Comment (jsmrcka) [reply http/graphql/src/test/java/io/quarkus/ts/http/graphql/GraphQLGetDisabledIT.java]: Added a POST query and renamed.

File: http/graphql/src/main/java/io/quarkus/ts/http/graphql/PersonsEndpoint.java [hunk 2]
```diff
@@ -44,16 +54,47 @@ public Person getPhilosopher(@Name("name") String name) {
 
     @Query("friend_r")
```
[the remark this answers is not in the document]
Comment (jsmrcka) [reply http/graphql/src/main/java/io/quarkus/ts/http/graphql/PersonsEndpoint.java]: Renamed and separated into reactive/non-reactive endpoints.

File: http/graphql/src/main/java/io/quarkus/ts/http/graphql/PersonsReactiveEndpoint.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import org.eclipse.microprofile.graphql.Mutation;
+import org.eclipse.microprofile.graphql.Name;
+import org.eclipse.microprofile.graphql.Query;
+
+import io.smallrye.graphql.api.Context;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+
+@GraphQLApi
+public class PersonsReactiveEndpoint extends PersonsEndpointBase {
```
Comment (fedinskiy) [inline http/graphql/src/main/java/io/quarkus/ts/http/graphql/PersonsReactiveEndpoint.java:14]: Just curious: is there an option to have separate endpoints?

Comment (jsmrcka) [reply http/graphql/src/main/java/io/quarkus/ts/http/graphql/PersonsReactiveEndpoint.java:14]: In general, yes. But I havent found support in Quarkus/Smallrye.

**Review by fedinskiy — APPROVED** [review]

**Comment by fedinskiy** [comment]
LGTM, but lets wait for CI.

And do we need three separate commits?

**Comment by jsmrcka** [comment]
> And do we need three separate commits?

First 2 just fix/modify existing stuff, last one adds new tests. It feels more organized. :smile:

**Comment by fedinskiy** [comment]
CI failed due to unrelated Kafka failure

**Pull request merged**